File: tests/fixtures/python/sample.py
# Sample Python Module

"""
This is a comprehensive Python test fixture.
It tests various language features for the analyzer.
"""

import os
import sys
from typing import List, Optional
from pathlib import Path

# TODO: Add error handling
# FIXME: Optimize this function

class DataProcessor:
    """Process data with various methods."""

    def __init__(self, name: str):
        self.name = name
        self.data = []

    def process(self, items: List[str]) -> int:
        """Process items and return count."""
        for item in items:
            self.data.append(item.upper())
        return len(self.data)

    @staticmethod
    def validate(value: str) -> bool:
        """Validate a value."""
        return len(value) > 0

@decorator_example
def decorated_function(x: int, y: int) -> int:
    """A decorated function."""
    result = x + y
    return result * 2

async def async_handler(data: bytes) -> Optional[str]:
    """Async function example."""
    decoded = data.decode('utf-8')
    return decoded if decoded else None

def main():
    """Main entry point."""
    processor = DataProcessor("test")
    count = processor.process(["a", "b", "c"])
    print(f"Processed {count} items")

if __name__ == "__main__":
    main()

File: tests/fixtures/rust/sample.rs
// Sample Rust Module

use std::io::{self, Write};
use std::collections::HashMap;

/// Configuration struct
pub struct Config {
    name: String,
    values: HashMap<String, i32>,
}

impl Config {
    pub fn new(name: String) -> Self {
        Config {
            name,
            values: HashMap::new(),
        }
    }

    pub fn set(&mut self, key: String, val: i32) {
        self.values.insert(key, val);
    }

    pub fn get(&self, key: &str) -> Option<&i32> {
        self.values.get(key)
    }
}

pub trait Processable {
    fn process(&self) -> Result<(), String>;
    fn validate(&self) -> bool;
}

pub async fn async_handler(data: Vec<u8>) -> Result<(), io::Error> {
    let processed: Vec<u8> = data.iter().map(|x| x * 2).collect();
    io::stdout().write_all(&processed)?;
    Ok(())
}

fn main() {
    let mut config = Config::new("app".to_string());
    config.set("port".to_string(), 8080);
    println!("Config ready");
}
